src/Components/Server/src/DependencyInjection/RazorComponentsServiceCollectionExtensions.cs

```diff
@@ -94,6 +94,8 @@ public static class RazorComponentsServiceCollectionExtensions
 
         private static void AddStandardRazorComponentsServices(IServiceCollection services)
         {
+            services.AddLogging();
```

Comment:
Is this safe to do even if the developer is explicitly adding their own logging service elsewhere in Startup.cs? I'm not super familiar with the internals of this.

Reply:
It should be, but I don't see MVC or SignalR doing this. I'll move this to the test app.

Reply:
It's likely they don't do because it's already added by something else they need like DataProtection. In any case, it's safe to do this, if it's not idempotent it's a bug in the method implementation. That is the intended behavior. 

You call what you need and it will noop if called more than once.